src/atlas/interpolation/method/bil/BilinearRemapping.cc

```diff
@@ -0,0 +1,395 @@
+                }
+                kpts *= 2;
+            }
+
+            if (!success) {
+                failures.push_back(ip);
+                if (not treat_failure_as_missing_value_) {
+                    Log::debug() << "------------------------------------------------------"
+                                    "---------------------\n";
+                    Log::debug() << "Failed to project point (lon,lat)=" << p << '\n';
+                    Log::debug() << failures_log.str();
+                }
+            }
+        }
+    }
+    Log::debug() << "Maximum neighbours searched was " << eckit::Plural(max_neighbours, "element") << std::endl;
+
+    if (failures.size()) {
+        if (treat_failure_as_missing_value_) {
+            missing_.resize(failures.size());
+            std::copy(std::begin(failures), std::end(failures), missing_.begin());
+        }
+        else {
+            // If this fails, consider lowering atlas::grid::parametricEpsilon
+            std::ostringstream msg;
+            msg << "Rank " << mpi::rank() << " failed to project points:\n";
+            for (std::vector<size_t>::const_iterator i = failures.begin(); i != failures.end(); ++i) {
+                const PointLonLat pll{(*olonlat_)(*i, (size_t)0), (*olonlat_)(*i, (size_t)1)};  // lookup point
+                msg << "\t(lon,lat) = " << pll << "\n";
+            }
+
+            Log::error() << msg.str() << std::endl;
+            throw_Exception(msg.str());
+        }
+    }
+
+    // fill sparse matrix and return
+    Matrix A(out_npts, inp_npts, weights_triplets);
+    matrix_shared_->swap(A);
+}
+
+Method::Triplets BilinearRemapping::projectPointToElements(size_t ip, const ElemIndex2::NodeList& elems,
+                                                           std::ostream& /* failures_log */) const {
+    ATLAS_ASSERT(elems.begin() != elems.end());
+
+    const size_t inp_points = ilonlat_->shape(0);
+    std::array<size_t, 4> idx;
+    std::array<double, 4> w;
+
+    Triplets triplets;
+    triplets.reserve(4);
+
+    double lon{(*olonlat_)(ip, 0)};
+    while (lon > 360.0) { lon -= 360.0; }
+    while (lon < 0.0) { lon += 360.0; }
+    PointXY ob_loc{lon, (*olonlat_)(ip, 1)};  // lookup point
+
+    idx_t single_point;
+    for (ElemIndex2::NodeList::const_iterator itc = elems.begin(); itc != elems.end(); ++itc) {
+        const idx_t elem_id = idx_t((*itc).value().payload());
+        ATLAS_ASSERT(elem_id < connectivity_->rows());
+
+        const idx_t nb_cols = connectivity_->cols(elem_id);
+        ATLAS_ASSERT(nb_cols == 3 || nb_cols == 4);
+
+        for (idx_t i = 0; i < nb_cols; ++i) {
+            idx[i] = (*connectivity_)(elem_id, i);
+            ATLAS_ASSERT(idx[i] < inp_points);
+        }
+
+        constexpr double tolerance = 1.e-12;
+
+        if (nb_cols == 3) {
+            /* triangle */
+            element::Triag2D triag(PointXY{(*ilonlat_)(idx[0], size_t(0)), (*ilonlat_)(idx[0], size_t(1))},
```

Comment:
You can omit the size_t constructors here

Reply:
thanks, I have removed in latest commit.